contracts/AppNFTUpgradeable.sol

```diff
@@ -70,6 +71,23 @@ contract AppNFTUpgradeable is Initializable, ERC721Upgradeable, ERC721Enumerable
     // flag to check if the minting is paid or not
     bool public payForMintFlag;
 
+    // structure that will store rating, timestamp & reviewLink
+    struct Review {
+        uint128 rating;
+        uint128 timestamp;
+        string reviewLink;
+    }
+    struct RatingData {
+        uint128 totalRating;
```

Comment:
- I think a more appropriate name would be `avgRating`. 
- This `avgRating` to be calculated on each rating addition. So that when it's being fetched no compute would be needed.